### ISSUE TITLE
`ty` reports unresolved import for `win32gui` despite successful import in Python

### ISSUE THREAD

**Issue by ooopus** [astral-sh/ty#530]
### Summary

`ty check` reports an `unresolved-import` error for the `win32gui` module, even though the module can be successfully imported and used in a Python environment where `pywin32` is installed.

**Steps to Reproduce**

1.  Install `pywin32`:
    ```bash
    pip install pywin32
    ```
2.  Create a Python file (e.g., `test.py`) with the following content:
    ```python
    import win32gui
    print("win32gui imported successfully")
    ```
3.  Verify that the Python script runs successfully and `win32gui` can be imported:
    ```bash
    python test.py
    ```
    Output:
    ```
    win32gui imported successfully
    ```
4.  Run `ty check` on the Python file:
    ```bash
    uvx ty check test.py
    ```

**Expected Behavior**

`ty check` should not report any errors, as `win32gui` is a valid and importable module after installing `pywin32`.

**Actual Behavior**

`ty check` reports an `unresolved-import` error:

```
WARN ty is pre-release software and not ready for production use. Expect to encounter bugs, missing features, and fatal errors.
error[unresolved-import]: Cannot resolve imported module `win32gui`
 --> test.py:1:8
  |
1 | import win32gui
  |        ^^^^^^^^
  |
info: make sure your Python environment is properly configured: https://github.com/astral-sh/ty/blob/main/docs/README.md#python-environment
info: rule `unresolved-import` is enabled by default

Found 1 diagnostic
```

### Version

ty 0.0.1-alpha.7 (afb20f6fe 2025-05-26)

**Comment by AlexWaygood** [comment]
Thanks for the report! Can I double-check two things:
1. Which environment are you installing `pywin32` into -- a virtual environment or your system installation of Python?
2. Do you have a virtual environment locally (if so, what directory is it in and have you activated it)?

**Comment by ooopus** [comment]
> Thanks for the report! Can I double-check two things:
> 
> 1. Which environment are you installing `pywin32` into -- a virtual environment or your system installation of Python?
> 2. Do you have a virtual environment locally (if so, what directory is it in and have you activated it)?

1. I have installed pywin32 in both the virtual environment and the system environment
2. An error occurs regardless of whether the virtual environment (.venv) is activated.

**Comment by AlexWaygood** [comment]
Ohhh, I think `win32gui` might be a C extension, if I remember correctly? You might need to install `types-pywin32` for us to be able to resolve the import. If that's the case, then this is #487.

**Comment by ooopus** [comment]
Indeed, that's so. Thank you!

**Comment by AlexWaygood** [comment]
No problem. Sorry our diagnostics aren't better here. We'll definitely try to improve this.
